sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -290,9 +291,12 @@ def __init__(self, n, compare=None, key=None, reverse=False):
   def _sort_buffer(self, buffer, lt):
     if lt in (operator.gt, operator.lt):
       buffer.sort(key=self._key_fn, reverse=self._reverse)
+    elif self._key_fn:
+      buffer.sort(key=cmp_to_key(
+          (lambda a, b: (not lt(self._key_fn(a), self._key_fn(b)))
+           - (not lt(self._key_fn(b), self._key_fn(a))))))
     else:
-      buffer.sort(cmp=lambda a, b: (not lt(a, b)) - (not lt(b, a)),
```

Comment:
@robertwb Is it intentional that `reverse` is used only in the first branch?

Reply:
yes, just as it was before: the reverse is included on line 286

Reply:
Yes, my question is not related to this PR, the PR preserves existing behavior, and we should merge it.

It was not intuitive to me why first branch we  pass `reverse` to the sort function, and in second branch we do not, so I was wondering if there was a bug in the code. Looking closer at  Line 273-285, seems like the code is doing the right thing.